test/e2e_appium/conftest.py

```diff
@@ -0,0 +1,260 @@
+import os
+import pytest
+from datetime import datetime
+from pathlib import Path
+
+from .config import setup_logging, log_test_start, log_test_end, get_config
+from .config.logging_config import get_logger
+from .core import EnvironmentSwitcher
+
+
+_logging_setup = None
+
+
+def pytest_configure(config):
+    global _logging_setup
+    _logging_setup = setup_logging()
+    
+    # Try to use new YAML-based configuration, fallback to old system
+    try:
+        env_name = os.getenv('CURRENT_TEST_ENVIRONMENT', 'lambdatest')
+        switcher = EnvironmentSwitcher()
+        env_config = switcher.switch_to(env_name)
+        
+        # Use directories from YAML config
+        reports_dir = Path(env_config.directories.get('reports', 'reports'))
+        enable_xml_report = env_config.logging_config.get('enable_xml_report', True)
+        enable_html_report = env_config.logging_config.get('enable_html_report', True)
+        
+        logger = get_logger('conftest')
+        logger.info(f"📁 Using reports directory from {env_name} config: {reports_dir}")
+        
+    except Exception as e:
+        # Fallback to legacy configuration
+        test_config = get_config()
+        reports_dir = Path(test_config.reports_dir)
+        enable_xml_report = test_config.enable_xml_report
+        enable_html_report = test_config.enable_html_report
+        
+        logger = get_logger('conftest')
+        logger.warning(f"⚠️ Using legacy configuration for reports: {e}")
```

Comment:
I'm trying to understand why we have legacy configuration in completely new framework, it can be very hard to debug in future, as you  have to maintain two code paths, increasing the risk of inconsistencies and errors. I'd say **fail fast and loud** is better approach here: If the new config fails, raise an error and provide a clear message about what’s wrong and how to fix it.